libraries/AP_Notify/Display.cpp

```diff
@@ -488,11 +488,24 @@ void Display::update_mode(uint8_t r)
 	}
 }
 
+void Display::update_text_empty(uint8_t r)
+{
+    char msg [DISPLAY_MESSAGE_SIZE];
+    memset(msg, ' ', sizeof(msg));
+    draw_text(COLUMN(0), ROW(0), msg);
```

Comment:
The draw_textm method makes the end of the message a NULL code. All are initialized with white space with memset.
Continue to do inappropriate access until it hits the NULL code.